packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -0,0 +1,175 @@
+// are all returning prices in the same precision as `decimals`.
+class BasketSpreadPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new BasketSpreadPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {List} baselinePriceFeeds The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {List} experimentalPriceFeed The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {Object?} denominatorPriceFeed We optionally divide the price spread between the baseline and experimental baskets by this denominator price
+   *      in order to "denominate" the basket spread price in a specified unit. For example, we might want to express the basket spread in terms
+   *      of ETH-USD.
+   */
+  constructor(web3, logger, baselinePriceFeeds, experimentalPriceFeeds, denominatorPriceFeed) {
+    super();
+
+    if (baselinePriceFeeds.length === 0 || experimentalPriceFeeds.length === 0) {
+      throw new Error("BasketSpreadPriceFeed cannot be constructed with empty baseline or experimental baskets.");
+    }
+
+    this.baselinePriceFeeds = baselinePriceFeeds;
+    this.experimentalPriceFeeds = experimentalPriceFeeds;
+    this.denominatorPriceFeed = denominatorPriceFeed;
+
+    // For convenience, concatenate all constituent price feeds.
+    this.allPriceFeeds = this.baselinePriceFeeds.concat(this.experimentalPriceFeeds);
+    if (this.denominatorPriceFeed) {
+      this.allPriceFeeds = this.allPriceFeeds.concat(this.denominatorPriceFeed);
+    }
+
+    // Helper modules.
+    this.web3 = web3;
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+    this.logger = logger;
+
+    // The precision that the user wants to return prices in must match all basket constituent price feeds and the denominator.
+    this.decimals = this.allPriceFeeds[0].getPriceFeedDecimals();
+
+    // Scale `number` by 10**decimals.
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), this.decimals).toString());
+    };
+  }
+
+  // Given lists of experimental and baseline prices, and a denominator price,
+  // return the spread price, which is:
+  // (avg(experimental) - avg(baseline) + 1) / denominator
+  _getSpreadFromBasketPrices(experimentalPrices, baselinePrices, denominatorPrice) {
+    // Compute experimental basket mean.
+    if (
+      experimentalPrices.length === 0 ||
+      experimentalPrices.some(element => element === undefined || element === null)
```

Comment:
Not sure what would cause an undefined element here, but you could use a lodash compact or filter(x=>x) to get rid of nulls in array to avoid the check

Reply:
Hm but you'd also have to do the `length === 0` check prior to the filter